trunk/SUAVE/Core/Data.py

```diff
@@ -284,7 +284,11 @@ def __str2(self,indent=''):
                         val = value.__str2(indent+new_indent)
                     except RuntimeError: # recursion limit
                         val = ''
-                        
+                    except:
+                        val = value.__str__(indent+new_indent)                        
+                    finally:
```

Comment:
I'm not very familiar with this, but can you explain the purpose of this finally clause? My understanding is that this means if an error is still not caught or is generated in the last except (which is this case would catch any previous errors) it will run before passing the error up the stack. This doesn't seem important with a pass statement.

Reply:
Finally clauses are used when all else fails.

Since this is a printing statement a pass will just continue with the printing. I've since removed it. But the initial thought was that if an object was somehow unprintable the process of printing would not stop, and it would continue printing.

Reply:
The finally clause would still throw the uncaught error after executing the pass statement. For example try running:
```
try:
    1/0
except ValueError:
    print 'except 1'
finally:
    print 'finally'
    pass

print 'done'
```
Anyway see my other comment on your other change.